app/scripts/metamask-controller.js

```diff
@@ -4460,21 +4560,35 @@ export default class MetamaskController extends EventEmitter {
           this.networkController.upsertNetworkConfiguration.bind(
             this.networkController,
           ),
-        setActiveNetwork: this.networkController.setActiveNetwork.bind(
-          this.networkController,
-        ),
+        setActiveNetwork: (networkClientId) => {
+          this.selectedNetworkController.setNetworkClientIdForMetamask(
```

Comment:
was this needed for queuing? or did we just miss this when adding SelectedNetworkController into extension?

Reply:
yeah it was missed. Well, sort of. 

Instead of doing this, we were lstening to networkchanges and settng the metamask selected network (inside the selected network controller).

This was causing a bug where I was unable to change the network manually.  decided to remove the listener and to more deliberately update the selected network.

Reply:
ahh i see.. whelp that's too bad. I feel like it made sense listening to it in the SelectedNetworkController. We have a weird hodgepodge of controllers that are affected by onXYZStateChanges directly and indirectly

Reply:
yeah maybe we can revisit this and try again